### DISCUSSION TITLE
can any one help to perform modelling of Floating PV Model using pvlib?

### DISCUSSION THREAD

**Discussion by hari0206** [pvlib/pvlib-python#1664] [Q&A]
can any one help to perform modelling of Floating PV Model using pvlib?
thanks in advance

**Comment by hari0206** [comment]
@wholmgren

**Comment by cwhanse** [comment]
@hari0206 I think we're all interested to see how this can be done, and to hear where capabilities in pvlib may need development. I haven't modeled a floating system, or vehicle-mounted PV, where the modules orientation relative to the sun may change frequently and not systematically. Let us know what you learn!

**Reply by kandersolar** [reply]
I have not done more than a brief skim, but this paper goes into a little bit of detail on modeling floating PV using pvlib-python: https://doi.org/10.1016/j.seta.2021.101515

Prinsloo et al. 2021, "Sustainability assessment framework and methodology with trans-disciplinary numerical simulation model for analytical floatovoltaic energy system planning assessments"

**Comment by AlbaAlcaniz** [comment]
@wholmgren @cwhanse 
I would like to contribute to pvlib with the code that I have developed and has been recently published in https://www.sciencedirect.com/science/article/pii/S0196890423012438
The code provides the tilt and azimuth of an offshore floating PV system in the North sea provided the wind speed data. However, the spectrum can be modified to simulate any other sea. Once the angles are obtained, they can be used as inputs to the other functions of pvlib.
The code is in matlab but I am willing to translate it to python. Please let me know your thoughts and if you agree I can open an issue about it and start working on it

**Comment by cwhanse** [comment]
@AlbaAlcaniz this is the work presented at the 2023 European PVPMC?

The Matlab code looks straightforward. I'm a little concerned about the data inputs needed for `sizes_and_weights.m`, I think we'll want to consider creating a class or other structure to bundle those physical parameters.

It would be nice to have a sample sea spectrum with the code.

For organization, code for floating PV needs its own module, I think - it doesn't fit into anything currently and is a specific use case. `floating.py` suggests itself.

@pvlib/pvlib-maintainer I'm in favor.

**Reply by AlbaAlcaniz** [reply]
Yes, it is the work that I presented at the 2023 European PVPMC.

I also thought that creating a class for the pontoon would be a good idea. 
And I agree that it doesn't fit anywhere, so the file `floating.py` is a nice suggestion.

I can think on how to organize the code, and then create an issue proposing my approach before starting working on it. Does that sound good?

**Reply by cwhanse** [reply]
It is a plan.
